libcxx/test/libcxx/numerics/numarray/assert.pass.cpp

```diff
@@ -0,0 +1,42 @@
+//===----------------------------------------------------------------------===//
```

Comment:
Note: I found it easier to just test all the assertions in a single file (we have precedent in `deque`). We could split by hardening mode later, but I think having a separate test file per hardened function is more trouble than it's worth. No strong feelings, though -- please feel free to push back on this.